Tests/ResponseTests.swift

```diff
@@ -672,8 +672,27 @@ final class ResponseTryMapErrorTestCase: BaseTestCase {
         else { XCTFail(); return }
 
         XCTAssertEqual(underlyingError.asAFError?.isSessionTaskError, true)
-        XCTAssertEqual((underlyingError.asAFError?.underlyingError as? URLError)?.code, .cannotFindHost)
-
+        XCTAssertEqual(underlyingError.asAFError?.isHostURLError, true)
         XCTAssertNotNil(response?.metrics)
     }
 }
+
+private extension DataResponse where Failure == AFError {
+    var hasHostURLError: Bool? {
+        return error?.isHostURLError
+    }
+}
+
+private extension AFError {
+    var isHostURLError: Bool {
+        return urlHostErrorCodes.contains(urlErrorCode)
+    }
+    
+    var urlErrorCode : URLError.Code {
```

Comment:
I don't think a separate property is necessary here, and it's not really appropriate to fall back to the `.unknown` case rather than using `nil`.